app/app.js

```diff
@@ -4,18 +4,29 @@ const app = express()
 
 app.get('/api/topics', getTopics)
 
+app.get('/api', )
+
 app.use((err, request, response, next) =>{
     console.log(err)
-    response.status(500).send({msg: "Internal server error"})
-  })
-
-
-
-
-
-
-app.listen(9090, () => {
-    console.log('listening on 9090')
-})
+    if (err.status && err.msg) {
+        res.status(err.status).send({ msg: err.msg });
+      }
+      else if (err.code === '22P02') {
+        res.status(400).send({ msg: err.message || 'Bad Request' });
+      } else {
+        console.log(err);
+        res.status(500).send({ msg: 'Internal Server Error' });
+      }
+    });
+
+
+    let server;
```

Comment:
This should absolutely not be here. You'll need a listen.js file when you come to host that is external to this file.